### DISCUSSION TITLE
Spreadsheet and files

### DISCUSSION THREAD

**Discussion by ToddChastain** [danielmiessler/Fabric#716] [Q&A]
I am trying to use spreadsheets and files and pipe them into a fabric pattern that I have created. For example, I want to take an Excel spreadsheet and pipe it into a pattern that I have created that will build a PowerPoint outline based on the data provided. This is something I would be able to, do with the UI of chat GPT using the prompt that I have created but ultimately I would like to use fabric. Does anyone know a way to be able to leverage existing spreadsheets and files and them into fabric and leverage patterns to accomplish tasks. 

I am using MacOS

**Comment by patonw** [comment]
I haven't tried this personally but I'm going to hazard a guess that they're converting the excel files to CSV or JSON under the hood, which you can do with pandas and openpyxl.
